article/app/services/dotcomponents/AMPPicker.scala

```diff
@@ -19,24 +21,29 @@ object AMPPageChecks extends Logging {
     !page.article.tags.isPaidContent
   }
 
-  def hasOnlySupportedElements(page: PageWithStoryPackage): Boolean = {
+  def hasOnlySupportedElements(page: PageWithStoryPackage, blocks: APIBlocks): Boolean = {
     // See: https://github.com/guardian/dotcom-rendering/blob/master/packages/frontend/amp/components/lib/Elements.tsx
-    def supported(block: BlockElement): Boolean = block match {
-      case _: TextBlockElement => true
-      case _: ImageBlockElement => true
-      case _: InstagramBlockElement => true
-      case _: TweetBlockElement => true
-      case _: RichLinkBlockElement => true
-      case _: CommentBlockElement => true
-      case _: PullquoteBlockElement => true
-      case _ => false
+    // And also PageElement.scala here.
+    // It is necessary to look at both to check that we handle the relevant element type here and also handle the
+    // resulting PageElement model in DCR.
+    def supported(element: ElementType): Boolean = element match {
```

Comment:
If I understand this correctly (LOL), this now uses the CAPI Block names - we've got more granularity in the DCR model, in that `Video` is split into multiple elements. Are we happy to block on having all elements of a category complete in this way? This is probably more if we use similar picker for Frontend than AMP although Atoms are split into types in DCR?

Reply:
Ah nevermind just seen https://github.com/guardian/frontend/pull/21368#discussion_r279765161 (not deleting the comment though for a decision reference!)

Reply:
Yeah, it's not ideal and we might need a better strategy for non-AMP, but given how far along with AMP we are it feels good enough!